python/tchannel/scheme.py

```diff
@@ -41,10 +62,18 @@ def __init__(self):
     def type(self):
         return self._type
 
-    def deserialize(self, obj):
+    def deserialize_header(self, obj):
+        assert obj
```

Comment:
Don't use `assert` outside of tests. If it's an error then raise an exception, or handle it. In this case I think it would be more appropriate for the JSON library to raise an exception that we can catch.